### PR TITLE
Issue 69 testing

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Add some tests for convergence as mesh size is decreased and as small parameter in the asymptotics tends to zero (some of the tests suggested in #69). Not all converging as expected.
Not a complete issue fix but sets out the framework. Should be easier to work on once `simplify` and `jacobian` are implemented (frustratingly slow atm)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_models/test_lead_acid/test_asymptotics_convergence.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        parameter_values.process_model(full_model)
+        geometry = full_model.default_geometry
+        parameter_values.process_geometry(geometry)
+
+        # Discretise (same mesh, create different discretisations)
+        var_pts = {
+            pybamm.standard_spatial_vars.x_n: 3,
+            pybamm.standard_spatial_vars.x_s: 3,
+            pybamm.standard_spatial_vars.x_p: 3,
+            pybamm.standard_spatial_vars.r_n: 1,
```
Comment (Scottmar93) [inline tests/test_models/test_lead_acid/test_asymptotics_convergence.py]: can remove r_n and r_p

Comment (tinosulzer) [reply tests/test_models/test_lead_acid/test_asymptotics_convergence.py]: that gives
```
KeyError: 'Points not given for a variable in domain negative particle'
```
